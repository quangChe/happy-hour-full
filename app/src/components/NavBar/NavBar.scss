#navigation-container {
  height: 100%;
  width: 100%;
  position: absolute;

  .navbar {
    position: absolute;
    display: flex;
    justify-content: space-between;
    z-index: 10;
    width: 100%;
    padding: 20px;
    background: rgba(0, 0, 0, 0);
    box-sizing: border-box;
    height: 60px;
    top: 30px;
  
    @media only screen and (min-width: 400px) {
      padding: 20px 50px;
    }
  }
}


