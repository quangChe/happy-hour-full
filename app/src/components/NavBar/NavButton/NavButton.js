import React from 'react';
import './NavButton.scss';

export default class NavButton extends React.Component {
  state = {
    open: this.props.open ? this.props.open : false,
  };

  componentWillReceiveProps(nextProps) {
    if (nextProps.open !== this.state.open) {
      this.setState({open: nextProps.open})
    }
  }

  handleClick = () => {
    this.setState({open: !this.state.open});
    this.props.onClick(!this.state.open);
  }

  render() {
    const styles = {
      container: {
        background: this.state.open ? 'transparent' : '#dad3b8',
      },
      line: {
        background: this.state.open ? '#dad3b8' : '#333333',
      },
      lineTop: {
        transform: this.state.open ? 'translateX(3px) rotate(45deg)' : 'none',
        transformOrigin: 'top left',
        marginBottom: '5px',
      },
      lineMiddle: {
        opacity: this.state.open ? 0 : 1,
        transform: this.state.open ? 'translateX(16px)' : 'none',
      },
      lineBottom: {
        transform: this.state.open ? 'translateY(5px) translateX(2px) rotate(-45deg)' : 'none',
        transformOrigin: 'top left',
        marginTop: '5px',
      }
    }

    return (
      <div className="nav-btn-container" style={styles.container}
        onClick={this.handleClick}>
        <div className="line" style={{...styles.line, ...styles.lineTop}}/>
        <div className="line" style={{...styles.line, ...styles.lineMiddle}}/>
        <div className="line" style={{...styles.line, ...styles.lineBottom}}/>
      </div>
    )
  }
}