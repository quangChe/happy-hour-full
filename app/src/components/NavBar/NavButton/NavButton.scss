.nav-btn-container {
  height: 40px;
  width: 40px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  padding: 4px;
  z-index: 3;
  border-radius: 50%;

  .line {
    height: 1px;
    width: 24px;
    transition: all 0.4s ease-in-out;
    border-radius: 5px;
  }
}