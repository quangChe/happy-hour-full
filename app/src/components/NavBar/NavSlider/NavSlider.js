import React from 'react';
import './NavSlider.scss';
// import { Link } from 'react-router-dom';

export default class NavSliderMenu extends React.Component {
  render() {
    const { opened } = this.props;
    const styles = ['nav-menu', opened ? null : 'nav-menu-close'].join(' ');

    return (
      <div className={styles}>
        <ul>
          <li>Popular Sips</li>
          <li>Saved Sips</li>
          <li>Profile</li>
          <li>Settings</li>
        </ul>
      </div>
    )
  }
}