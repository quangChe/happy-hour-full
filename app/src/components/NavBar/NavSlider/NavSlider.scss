.nav-menu {
  position: fixed;
  height: 100%;
  width: 100%;
  background: #333333;
  transition: all 0.2s ease-in;
  transform: translateX(0);
  transform-origin: 100% 0;
  z-index: 10;
  display: flex;
  align-items: center;
  justify-content: center;
  overflow: hidden;

  ul {
    list-style: none;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    height: 30%;
    text-align: center;
    font-size: 2rem;
    font-weight: 700;
    color: #e6e6e6;

    li:hover {
      cursor: pointer;
      animation: color-change 1s infinite;
    }
  }
}

.nav-menu-close {
  transition: all 0.2s ease-out;
  transform: translateX(100%);
  transform-origin: 100% 0%;
}