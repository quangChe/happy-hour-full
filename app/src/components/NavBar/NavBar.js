import React from 'react';
import colors from '../../styles/colors';
import './NavBar.scss';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import Logo from '../Utils/Logo';
import NavButton from './NavButton/NavButton';
import NavSlider from './NavSlider/NavSlider';

export class NavBar extends React.Component {

  state = {
    menuOpen: false,
  }

  openNavMenu = (open) => {
    console.log(open);
    this.setState({menuOpen: open})
  }

  renderLogo = () => {
    const { menuOpen } = this.state;
    return menuOpen 
      ? (<Logo/>)
      : (<Link to="/" style={{textDecoration: 'none'}}><Logo/></Link>);
  }

  render() {
    const { loading } = this.props;
    const { menuOpen } = this.state;

    return loading
      ? null
      : (
        <div id="navigation-container">
          <NavSlider opened={menuOpen}/>
          <div className="navbar" style={{color: colors.black}}>
            {this.renderLogo()}
              <NavButton onClick={(e) => this.openNavMenu(e)} open={false}/>
          </div>
        </div>
      );
  }
}

const mapStateToProps = ({businesses: {nearbyBusinesses}}) => {
  return {loading: nearbyBusinesses.loading};
}

export default connect(mapStateToProps)(NavBar);