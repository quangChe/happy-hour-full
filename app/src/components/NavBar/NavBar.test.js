import React from 'react';
import { shallow } from 'enzyme';

import { NavBar } from './NavBar';
import NavSlider from './NavSlider/NavSlider';
import NavButton from './NavButton/NavButton';

describe('NavBar', () => {
  const wrapper = shallow(<NavBar  />);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();    
  })

  describe('while the app is still loading', () => {
    beforeEach(() => {
      wrapper.setProps({loading: true});    
    })

    it('does not render anything', () => {
      expect(wrapper.find(NavButton)).toHaveLength(0);
      expect(wrapper.find(NavSlider)).toHaveLength(0);
    });
  });

  describe('once app is done loading', () => {
    beforeEach(() => {
      wrapper.setProps({loading: false});  
    });

    it('renders all components', () => {
      expect(wrapper.find(NavButton)).toHaveLength(1);
      expect(wrapper.find(NavSlider)).toHaveLength(1);
    });

    describe('whenever user clicks the NavButton', () => {
      
      it('changes the `menuOpen` state to true if NavButton is opening slide-out menu', () => {
        wrapper.find(NavButton).props().onClick(true);
        expect(wrapper.state().menuOpen).toBe(true);
      });

      it('changes the `menuOpen` state to false if NavButton is closing slide-out menu', () => {
        wrapper.find(NavButton).props().onClick(false);
        expect(wrapper.state().menuOpen).toBe(false);
      });
    })
  });
})
