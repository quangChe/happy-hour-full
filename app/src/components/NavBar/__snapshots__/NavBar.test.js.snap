// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<div
  id="navigation-container"
>
  <NavSliderMenu
    opened={false}
  />
  <div
    className="navbar"
    style={
      Object {
        "color": "#333333",
      }
    }
  >
    <Link
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/"
    >
      <Logo />
    </Link>
    <NavButton
      onClick={[Function]}
      open={false}
    />
  </div>
</div>
`;
